File: Fortran/DateOfEaster.f90
!   Date of Easter
!   Based on algorithm by Jean Meeus, Astronomical Algorithms, 1998, p. 67 

program DateOfEaster
implicit none
    integer :: x, a, b, c, d, e, f, g, h, i, k, l, m, n, p, z

10  print *,'Enter year:' 
    read *, x
    
    if (x < 1583) then
        print *,'Year earlier than 1583 is not allowed!'
        go to 10
    endif

    a = mod(x, 19)
    b = x / 100
    c = mod(x, 100)
    d = b /4
    e = mod(b, 4)
    f = (b + 8) / 25;
    g = (b - f + 1) /3;
    h = mod(19 * a + b - d - g + 15, 30)
    i = c / 4
    k = mod(c, 4)
    l = mod(32 + 2 * e + 2 * i - h - k, 7)
    m = (a + 11 * h + 22 * l) / 451
    z = h + l - 7 * m + 114
    n = z / 31
    p = mod(z, 31)

    if (n == 3 .or. n == 4) then
        if (n == 3) then
            print *, "March ", p + 1
        else
            print *, "April ", p + 1
        endif
    endif

end program DateOfEaster

File: Perl/DateOfEaster.pl
#!/usr/bin/perl

# Date of Easter
# Based on algorithm by Jean Meeus, Astronomical Algorithms, 1998, p. 67

use strict;
use warnings;

my $x;
while (1)
{
    print "Enter a year:";
    $x = <STDIN>;
    chomp $x;
    if ($x >= 1583)
    {
        last;
    }
    print "Year earlier than 1583 is not allowed!\n";
}
 
my $a = $x % 19;
my $b = int($x / 100);
my $c = $x % 100;
my $d = int($b / 4);
my $e = $b % 4;
my $f = int(($b + 8) / 25);
my $g = int(($b - $f + 1) / 3);
my $h = (19 * $a + $b - $d - $g + 15) % 30;
my $i = int($c / 4);
my $k = $c % 4;
my $l = (32 + 2 * $e + 2 * $i - $h - $k) % 7;
my $m = int(($a + 11 * $h + 22 * $l) / 451);
my $z = $h + $l - 7 * $m + 114;
my $n = int($z / 31);
my $p = $z % 31 + 1;

if ($n == 3)
{
    print "March $p\n";
}
elsif ($n == 4)
{
    print "April $p\n";
}
